.github/workflows/check-md-link.yml

```diff
@@ -0,0 +1,43 @@
+#
+# Copyright (c) 2024 Snowflake Computing Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# This workflow uses actions that are not certified by GitHub.
+# They are provided by a third-party and are governed by
+# separate terms of service, privacy policy, and support
+# documentation.
+# This workflow will find markdown links in the folder paths
+# specified, and confirm the links are valid
+
+name: Check Markdown links
+
+on:
+  push:
+    paths:
+      - docs/**
+      - regtests/client/python/docs/**
```

Comment:
question: should we check the READM.md both in the root dir and in the dir `regtests`?

Reply:
good catch, and that's a good idea.  I used the Python code below to find all .md files in the repo, and found a couple I'm about to add to the checker:
- top-level .md's (the ones w/ links are README.md, CODE_OF_CONDUCT.md, CONTRIBUTING.md)
- regtests/README.md
- regtests/client/python/README.md

```python
from pathlib import Path
for file in Path('.').glob('**/*.md'):
    print(file)
```

Files not covered by current parameters
```text
CCLA.md
CODE_OF_CONDUCT.md
README.md
CONTRIBUTING.md
ICLA.md
SECURITY.md
regtests/README.md
regtests/client/python/README.md
.github/pull_request_template.md
```

Files covered by current parameters:
```text
regtests/client/python/docs/** (154 files)
docs/** (6 files)
```

Reply:
Thanks for doing this. I wish the github action can do the same thing as your python script does :-), so that we don't have to specify these paths. Ideally, the file type like *.md, or *.html would be the only input we need. The PR looks good to me though. Will merge it a bit later.

Reply:
Looks like by default, not specifying the `folder-path` param will cause the action to look at all the MD files in the repo. 
https://github.com/gaurav-nelson/github-action-markdown-link-check?tab=readme-ov-file#custom-variables

Maybe we can run this check for all MD files and for all PRs to cover all the edge cases. This way we don't need to allowlist new MD files not currently specified